commit 67e061f044e46c9823e59ca7981786f858bfa292
Merge: a3f132df0e5f b65054597872
Author: Mauro Carvalho Chehab <mchehab+huawei@kernel.org>
Date:   Tue Dec 1 16:21:40 2020 +0100

    Merge tag 'v5.10-rc6' into patchwork
    
    Linux 5.10-rc6
    
    * tag 'v5.10-rc6': (1815 commits)
      Linux 5.10-rc6
      sock: set sk_err to ee_errno on dequeue from errq
      mptcp: fix NULL ptr dereference on bad MPJ
      net: openvswitch: fix TTL decrement action netlink message format
      perf probe: Change function definition check due to broken DWARF
      perf probe: Fix to die_entrypc() returns error correctly
      perf stat: Use proper cpu for shadow stats
      perf record: Synthesize cgroup events only if needed
      perf diff: Fix error return value in __cmd_diff()
      perf tools: Update copy of libbpf's hashmap.c
      x86/mce: Do not overwrite no_way_out if mce_end() fails
      kvm: x86/mmu: Fix get_mmio_spte() on CPUs supporting 5-level PT
      KVM: x86: Fix split-irqchip vs interrupt injection window request
      KVM: x86: handle !lapic_in_kernel case in kvm_cpu_*_extint
      usb: typec: stusb160x: fix power-opmode property with typec-power-opmode
      printk: finalize records with trailing newlines
      can: af_can: can_rx_unregister(): remove WARN() statement from list operation sanity check
      can: m_can: m_can_dev_setup(): add support for bosch mcan version 3.3.0
      can: m_can: fix nominal bitiming tseg2 min for version >= 3.1
      can: m_can: m_can_open(): remove IRQF_TRIGGER_FALLING from request_threaded_irq()'s flags
      ...

diff --cc drivers/media/platform/mtk-vcodec/mtk_vcodec_enc_drv.c
index a2aa96b03f66,3be8a04c4c67..dfb42e19bf81
--- a/drivers/media/platform/mtk-vcodec/mtk_vcodec_enc_drv.c
+++ b/drivers/media/platform/mtk-vcodec/mtk_vcodec_enc_drv.c
@@@ -284,9 -284,16 +284,9 @@@ static int mtk_vcodec_probe(struct plat
  		mtk_v4l2_err("Could not get venc IPI device");
  		return -ENODEV;
  	}
 -	if (!pdev->dev.dma_parms) {
 -		pdev->dev.dma_parms = devm_kzalloc(&pdev->dev,
 -						sizeof(*pdev->dev.dma_parms),
 -						GFP_KERNEL);
 -		if (!pdev->dev.dma_parms)
 -			return -ENOMEM;
 -	}
 -	dma_set_max_seg_size(&pdev->dev, DMA_BIT_MASK(32));
 +	dma_set_max_seg_size(&pdev->dev, UINT_MAX);
  
- 	dev->fw_handler = mtk_vcodec_fw_select(dev, fw_type, VPU_RST_ENC);
+ 	dev->fw_handler = mtk_vcodec_fw_select(dev, fw_type, ENCODER);
  	if (IS_ERR(dev->fw_handler))
  		return PTR_ERR(dev->fw_handler);